specification/app/data-plane/Microsoft.App.DynamicSessions/preview/2024-10-02-preview/DynamicSessions.json

```diff
@@ -0,0 +1,797 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft.App Dynamic Sessions service",
+    "version": "2024-10-02-preview",
+    "x-typespec-generated": [
+      {
+        "emitter": "@azure-tools/typespec-autorest"
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{endpoint}",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "name": "endpoint",
+        "in": "path",
+        "description": "The management endpoint of the session pool.",
+        "required": true,
+        "type": "string"
+      }
+    ]
+  },
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "AadOauth2Auth": [
+        "https://dynamicsessions.io/.default"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "AadOauth2Auth": {
+      "type": "oauth2",
+      "description": "The Azure Active Directory OAuth2 Flow",
+      "flow": "accessCode",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "https://dynamicsessions.io/.default": ""
+      },
+      "tokenUrl": "https://login.microsoftonline.com/common/oauth2/token"
+    }
+  },
+  "tags": [],
+  "paths": {
+    "/executions": {
+      "post": {
+        "operationId": "CodeExecution_ExecuteCode",
+        "description": "Execute code in a session.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SessionIdentifier"
+          },
+          {
+            "$ref": "#/parameters/ExecutionOperationIdHeader"
+          },
+          {
+            "name": "codeExecutionRequest",
+            "in": "body",
+            "description": "The request to execute code.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SessionCodeExecutionRequest"
+            }
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
only 202 is allowed to return? what about 200?

Reply:
This follows [this guideline](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#patterns-to-initiate-a-long-running-operation) to only return 202 not 200 for LRO as a previous [comment](https://github.com/Azure/azure-rest-api-specs/pull/30424#discussion_r1777915461) says